torch/ao/quantization/fx/convert.py

```diff
@@ -174,7 +195,32 @@ def _replace_observer_with_quantize_dequantize_node(
         quantize_op = "to"
         qparams = {"_dtype_": dtype}
 
-    # 2. replace observer node with quant - dequant node
+    # 2. insert choose_qparams op and update the params list for the is_decomposed and
+    # is_dynamic case
+    if is_decomposed and is_dynamic:
```

Comment:
it's a bit weird that in the previous if block we have

```
elif compute dtype in (...):
  if is_dynamic:
    ...
  
```

and in this one we have

```
if is_decomposed and is_dynamic:
  ...
```

can we keep it consistent?

Reply:
sure

Reply:
it's just a special case for is_decomposed + is_dynamic btw, we don't have other cases, at least for now, but I don't have a strong preference on this

Reply:
actually this feels a bit weird as well when I look at it...:
```
if is_decomposed:
    if is_dynamic:
        ...

# some other code
...
```